File: bite_373.js
function reverseOnlyLetters(input) {
  let ans = Array.from(input);
  let i = 0;
  let j = ans.length - 1;

  while (i < j) {
    while (i < j && !/[a-zA-Z]/.test(ans[i])) {
      i++;
    }
    while (i < j && !/[a-zA-Z]/.test(ans[j])) {
      j--;
    }
    [ans[i], ans[j]] = [ans[j], ans[i]];
    i++;
    j--;
  }

  return ans.join('');
}

File: bite_373.py
''' Bite 373. Reverse only Letters
Given a string, reverse only the English letters (lowercase or uppercase) in the string, leaving all non-English letters and other characters (e.g. numbers) in their original position.'''
# Online Python - IDE, Editor, Compiler, Interpreter
def reverse_only_letters(input):
    ans = list(input)
    i = 0
    j = len(input) - 1

    while i < j:
        while i < j and not input[i].isalpha():
            i += 1
        while i < j and not input[j].isalpha():
            j -= 1
        ans[i], ans[j] = ans[j], ans[i]
        i += 1
        j -= 1
    ans = ''.join(ans)
    return ans

print(reverse_only_letters("ab-cd"))
print(reverse_only_letters("ab5DEf"))
print(reverse_only_letters("a-bC-dEf-ghIj"))
